Nlp explanation

kodymoodley opened the pull request:
This PR adds features to the propose connections page which helps the user validate the proposed connections by the NLP algorithm.

**Features**
1. Adds table to the propose connections page with the list of user-specified connections. This creates an integrated view of all connections (proposed and user-specified).
2. In the proposed connections table, color highlighting is added for tokens that are involved in the semantic text matching process to determine if a connection should exist or not.

kodymoodley committed:
adds user-specified connections table to propose connections page

kodymoodley committed:
removes editing features for user-specified connections table in propose connections page

kodymoodley committed:
corrects typo

kodymoodley committed:
updates connections schema to include info about which items in statements match

kodymoodley committed:
implements retrieval of specific matched terms; updates testing to be compatible with new connections data format; formats code

kodymoodley committed:
implements color highlighting for matching tokens across statements

kodymoodley committed:
fixes incorrect logic of connection matching tokens highlighting

kodymoodley committed:
comments out failing testcases to pass build tests; updates help page

sverhoeven reviewed (changes requested):
After correcting load example 2 as described in https://github.com/ESI-FAR/INA-tool/pull/171#issuecomment-3043669638 I was able to get suggested connections.

The existing connections table renders correctly.

![localhost_5173__project=Example+2 (1)](https://github.com/user-attachments/assets/4d83761f-4be5-42ae-9a66-02f9ac33cf96)
I like how bold and background color can overlap.

This PR could be improved by inline suggestions.

kodymoodley commented:
Thanks for the thorough list of suggestions @sverhoeven. Will work on them this week.

kodymoodley committed:
fixes loadExample issues and bug with highlighting for inspections example network

kodymoodley committed:
fixes issues with read only existing connections table

kodymoodley committed:
fixes redundant type definition issue

kodymoodley committed:
fixes issues with StatementCell re-rendering and conversion of highlighting to React component

kodymoodley committed:
moves pruning of connection data to test code; updates e2e test case result to match updated load example dataset

kodymoodley commented on src/nlp/testdata/mitigationTestData.ts:
```diff
@@ -256,13 +256,11 @@ export const outcomeConnections: Connection[] = [
-  //   target_component: "Activation Condition",
-  //   driven_by: "outcome",
-  // },
+  {
```
Addressed in [2fcf0d5](https://github.com/ESI-FAR/INA-tool/pull/171/commits/2fcf0d510c36bc5ae3e9a9cf2ae48c9fd2fc3d8f)

kodymoodley commented on src/components/ReadOnlyDrivenConnectionTable.tsx:
```diff
@@ -0,0 +1,308 @@
+  );
+}
+
+function FilterSelect({
```
Addressed in [073d727](https://github.com/ESI-FAR/INA-tool/pull/171/commits/073d727acb0d87fbd250f28e248050d91f5e71d6)

kodymoodley commented on src/components/ReadOnlyDrivenConnectionTable.tsx:
```diff
@@ -0,0 +1,308 @@
+import { Link } from "@tanstack/react-router";
+
+const columns: ColumnDef<ConnectionWithValues>[] = [
+  selectColumnDefinition(),
```
Fixed in [073d727](https://github.com/ESI-FAR/INA-tool/pull/171/commits/073d727acb0d87fbd250f28e248050d91f5e71d6)

kodymoodley commented on src/components/StatementCell.tsx:
```diff
@@ -4,85 +4,233 @@ import { cn } from "@/lib/utils";
+    // Build segments
+    nonOverlappingMatches.forEach((match, index) => {
+      // Add text before the match
+      console.log(index);
```
Addressed in [54c24bc](https://github.com/ESI-FAR/INA-tool/pull/171/commits/54c24bc57e830ec71075bcbeb4bee15ffff7c04b)

kodymoodley commented:
@sverhoeven I have addressed all your comments in the latest commits above. The most interesting was the load example one where I did not even get any proposed connections for Example 1. Not sure if you also experienced this? I therefore had to slightly modify one of the statements for Example 1 to get a proposed connection to appear, and of course any resulting test case results dependent on this dataset. I also added two additional examples, one for Mitigations network and the other is the Inspections network (both given by Amineh via email). Let me know if this is ready to merge.

sverhoeven reviewed (approved):
Thanks for resolving my requests for change.

For me its good to merge, but let us wait for review by the end-user.

kodymoodley committed:
updates mitigation test statements to match Fig B.3 more closely

kodymoodley committed:
fixes missing negation check for actor connection finding

kodymoodley committed:
enriches wordnet dictionary to handle new cases

kodymoodley committed:
makes percentage of tokens to match threshold configurable and sets it to 100% by default (to pass tests for Fig B.3 actor connections)

kodymoodley committed:
removes matching of dispersed characters which causes some false positive connections

kodymoodley committed:
adds some cases to wordnet relations; cleans up test data

kodymoodley committed:
updates formatting and linting

sverhoeven reviewed (approved):
Looks good to me.

Only suggestion is to explain the 3 and 4 examples.

kodymoodley committed:
adds and corrects descriptions of example networks

kodymoodley committed:
updates linting

kodymoodley commented on src/components/LoadExampleButton.tsx:
```diff
@@ -179,6 +188,12 @@ export function LoadExampleButton() {
           <DropdownMenuItem onClick={loadExample2}>
             Load example 2
           </DropdownMenuItem>
+          <DropdownMenuItem onClick={loadExample3}>
```
Thanks @sverhoeven, I addressed this in [96ddb36](https://github.com/ESI-FAR/INA-tool/pull/171/commits/96ddb36fde5dacbfb8482a5ed8e8ea391c238a0e)
